x-pack/plugins/security_solution/public/resolver/models/database_parameters.ts

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+import { DatabaseParameters } from '../types';
+
+export function equal(param1: DatabaseParameters, param2?: DatabaseParameters) {
```

Comment:
Why add this extra abstraction layer? For the type checking?

Reply:
I think since timeline now sends resolver the ID and the indices we figured it'd be a good idea to combine them in a single object. Since `indices` is a string array we need some way to compare all the strings easily so Rob and I came up with this method. We might switch out the lodash `equal` though.

Reply:
I was specifically talking about the abstraction of `equal` over `isEqual`. I understand the combined type, that makes sense. It's really more, if this equality check is going to be focused solely on checking the equality of `DatabaseParameters`, we know the shape, so might as well explicitly check the equality of the individual fields?

Reply:
Oh you mean like, why not just use `_.isEqual` instead of our `equal` function? Or are you suggesting doing the comparisons directly in the places we're calling `equal`? I think @oatkiller was considering removing the lodash use so we'd have to implement an array comparison in that case. We'd probably want the array comparison at a minimum to be in a helper function because we do the `DatabaseParameters` check in like 3 different places I think.

I guess another option is we make `DatabaseParameters` into a class and move `equal` as a method of that class?

Reply:
Yea, that was my thought, but since it's going to be something different from `isEqual`, then I would just leave it. Depending on what it is, it could still be more generalized into a util, but no need to do that here anyways